src/routes/relay/relay.controller.spec.ts

```diff
@@ -0,0 +1,1646 @@
+                return Promise.resolve({ data: safe, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+          networkService.post.mockImplementation((url) => {
+            switch (url) {
+              case `${relayUrl}/relays/v2/sponsored-call`:
+                return Promise.resolve({ data: { taskId }, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+
+          await request(app.getHttpServer())
+            .post(`/v1/chains/${chain.chainId}/relay`)
+            .send({
+              to: safeAddress,
+              data,
+            })
+            .expect(201)
+            .expect({
+              taskId,
+            });
+        });
+
+        // swapOwner (execTransaction)
+        it('should return 201 when making a swapOwner call', async () => {
+          const chainId = faker.helpers.arrayElement(supportedChainIds);
+          const chain = chainBuilder().with('chainId', chainId).build();
+          const safe = safeBuilder().build();
+          const safeAddress = getAddress(safe.address);
+          const data = execTransactionEncoder()
+            .with('to', safeAddress)
+            .with('data', swapOwnerEncoder().encode())
+            .encode() as Hex;
+          const taskId = faker.string.uuid();
+          networkService.get.mockImplementation((url) => {
+            switch (url) {
+              case `${safeConfigUrl}/api/v1/chains/${chainId}`:
+                return Promise.resolve({ data: chain, status: 200 });
+              case `${chain.transactionService}/api/v1/safes/${safeAddress}`:
+                // Official mastercopy
+                return Promise.resolve({ data: safe, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+          networkService.post.mockImplementation((url) => {
+            switch (url) {
+              case `${relayUrl}/relays/v2/sponsored-call`:
+                return Promise.resolve({ data: { taskId }, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+
+          await request(app.getHttpServer())
+            .post(`/v1/chains/${chain.chainId}/relay`)
+            .send({
+              to: safeAddress,
+              data,
+            })
+            .expect(201)
+            .expect({
+              taskId,
+            });
+        });
+
+        // execTransaction (execTransaction)
+        it('should return 201 calling execTransaction on a nested Safe', async () => {
+          const chainId = faker.helpers.arrayElement(supportedChainIds);
+          const chain = chainBuilder().with('chainId', chainId).build();
+          const safe = safeBuilder().build();
+          const safeAddress = getAddress(safe.address);
+          const data = execTransactionEncoder()
```

Comment:
This is not intended to be tackled in this PR, it's more for a follow-up I guess, but it seems all the above test cases share the same setup except for the `data` property (depending on the transaction type). Not sure it is totally possible, but this seems a good candidate for parametrising the tests with `it.each`, by iterating on the `data` parameter. What do you think?

Reply:
Refactored in dc73902b4838bcb57f7a97a74f997cc572677a30.

Reply:
Thanks! 🙏🏻